src/main/java/com/enderio/base/client/model/PaintingQuadTransformer.java

```diff
@@ -0,0 +1,41 @@
+package com.enderio.base.client.model;
+
+import com.enderio.core.client.RenderUtil;
+import com.enderio.core.data.model.EIOModel;
+import net.minecraft.client.Minecraft;
+import net.minecraft.client.renderer.RenderType;
+import net.minecraft.client.renderer.block.model.BakedQuad;
+import net.minecraft.client.renderer.texture.TextureAtlasSprite;
+import net.minecraft.client.resources.model.BakedModel;
+import net.minecraft.core.Direction;
+import net.minecraft.util.RandomSource;
+import net.minecraft.world.level.block.state.BlockState;
+import net.minecraftforge.client.model.IQuadTransformer;
+import net.minecraftforge.client.model.data.ModelData;
+
+import java.util.List;
+
+public record PaintingQuadTransformer(BlockState paint, RenderType type) implements IQuadTransformer {
```

Comment:
If this is what I think it is; is there any way to make the purple and grey box things render with a nicer texture?

Reply:
Resolved in discord, i didn't understand this part :P